source/extensions/transport_sockets/tls/context_impl.cc

```diff
@@ -312,6 +312,15 @@ ContextImpl::ContextImpl(Stats::Scope& scope, const Envoy::Ssl::ContextConfig& c
           absl::StrCat("Failed to load certificate chain from ", ctx.cert_chain_file_path_));
     }
 
+    // The must staple extension means the certificate promises to carry
+    // with it an OCSP staple. https://tools.ietf.org/html/rfc7633#section-6
+    absl::string_view tls_feature_ext = "1.3.6.1.5.5.7.1.24";
+    absl::string_view must_staple_ext_value = "\x02\x01\x05";
+    auto must_staple = Utility::getX509ExtensionValue(*ctx.cert_chain_, tls_feature_ext);
+    if (must_staple.has_value() && must_staple.value() == must_staple_ext_value) {
+      ctx.is_must_staple_ = must_staple.has_value();
```

Comment:
`ctx.is_must_staple_ = true;`? I don't think `must_staple.has_value()` can be false here.